lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/YarnClasspathProvider.java

```diff
@@ -153,6 +156,14 @@
     return this.classPathSuffix;
   }
 
+  private void logEnvVariable() {
+    if (LOG.isLoggable(CLASSPATH_LOG_LEVEL)) {
+      Map<String, String> map = System.getenv();
```

Comment:
make variable final - or, better yet, just write 
`for(final Map.Entry<String, String> entry : System.getenv().entrySet())`